src/spinel_hdlc.cpp

```diff
@@ -210,7 +210,14 @@ otError HdlcInterface::Write(const uint8_t *aFrame, uint16_t aLength)
 
     while (aLength)
     {
-        ssize_t rval = write(mUartFd, aFrame, aLength);
+        ssize_t rval;
+
+        // Configure ESP-IDF UART to never convert "\n" to "\r\n" just before
+        // writing radio frames through UART. This is a workaround for issue:
+        // https://github.com/openthread/ot-esp32/issues/5.
+        esp_vfs_dev_uart_set_tx_line_endings(ESP_LINE_ENDINGS_LF);
+        rval = write(mUartFd, aFrame, aLength);
+        esp_vfs_dev_uart_set_tx_line_endings(ESP_LINE_ENDINGS_CRLF);
```

Comment:
I know this is a temporary workaround, but will it look better to set ESP_LINE_ENDINGS_CRLF/ESP_LINE_ENDINGS_LF every time before/after CLI Uart is written to?

Reply:
It seems possible to do that, will try.

Reply:
@lanyuwen Tried and it seems that changing line-ending configuration for every write at CLI UART requires changing underlying `vprintf` of ESP_LOG as well (ESP_LOG writes to `stdout` and `stderr` which is CLI UART by default). though this is doable with `esp_log_set_vprintf(...)`, but it introduce unnecessary complexity since CLI UART (UART 0) could be shared by more potential components/applications which are out of our control.

So manually configuring RADIO UART seems to be more practical as we own it exclusively.

Reply:
Ok, that's fine to proceed with the original fix if the new proposal introduces too much complexity.